### DISCUSSION TITLE
Issue with idl to code in Java

### DISCUSSION THREAD

**Discussion by GetIt-now** [OpenDDS/OpenDDS#4493] [Q&A] [answered]
I'm going to build idl file to java .jar file through the mwc script,
after I execute $ACE_ROOT/bin/mwc.pl -type gnuace -features java=1 command, make, have this error, can not find org.omg.CORBA.Object, use java_pre_jpms=0 still not work
![image](https://github.com/OpenDDS/OpenDDS/assets/65381731/cdab1fe6-c8e8-41d8-b776-abb511895755)

**Comment by mitza-oci** [comment]
From the DDS_ROOT directory, please run `tools/scripts/show_build_config.pl` and post the output as plain text here.

**Comment by GetIt-now** [comment]
Here is the result

```
:~$ $DDS_ROOT/tools/scripts/show_build_config.pl 
=================================== setenv.sh ==================================
# OpenDDS configure script: ./configure --java
export ACE_ROOT="/opt/ACE_wrappers"
export CIAO_ROOT="unused"
export DANCE_ROOT="unused"
export DDS_ROOT="/opt/OpenDDS-3.26.1"
export JAVA_HOME="/usr/lib/jvm/java-11-openjdk-amd64"
export JAVA_PLATFORM="linux"
export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/opt/ACE_wrappers/lib:/opt/OpenDDS-3.26.1/lib"
export MPC_ROOT="/opt/ACE_wrappers/MPC"
export PATH="${PATH}:/opt/ACE_wrappers/bin:/opt/OpenDDS-3.26.1/bin"
export TAO_ROOT="/opt/ACE_wrappers/TAO"
================================ user_macros.GNU ===============================
java_pre_jpms=0
java=1
================================== VERSION.txt =================================
This is OpenDDS version 3.26.1, released Nov 14 2023
================================== ACE Version =================================
#define ACE_VERSION "7.1.2"
============================== platform_macros.GNU =============================
include $(ACE_ROOT)/include/makeinclude/platform_linux.GNU
=================================== config.h ===================================
#include "ace/config-linux.h"
```

**Reply by mitza-oci** [reply]
I'm not sure why this isn't being handled automatically with the default.features file, but try adding `-features java_pre_jpms=0` to the `mwc.pl` command line.

**Comment by GetIt-now** [comment]
Thank you for your response, but there still have some problems,
I execute below command have error
$ACE_ROOT/bin/mwc.pl -type gnuace -features java_pre_jpms=0 java=1
Error Info:
Using .../opt/ACE_wrappers/bin/MakeProjectCreator/config/MPC.cfg
Generating 'gnuace' output using java=1
java=1: line 0:
ERROR: Unable to open "java=1" for reading
ERROR: Unable to process: java=1

Execute 
$ACE_ROOT/bin/mwc.pl -type gnuace -features java_pre_jpms=0     have below result:
gnuace -features java_pre_jpms=0
Using .../opt/ACE_wrappers/bin/MakeProjectCreator/config/MPC.cfg
Generating 'gnuace' output using default input
Skipping messenger_idl_test (messenger_idl_test.mpc); it requires java.
Generation Time: 0s
![image](https://github.com/OpenDDS/OpenDDS/assets/65381731/412acd0d-1f1c-45c2-bb15-fd629a8026ed)

**Reply by iguessthislldo** [reply]
`-features` is comma delimited. Try `$ACE_ROOT/bin/mwc.pl -type gnuace -features java_pre_jpms=0,java=1`

**Reply by mitza-oci** [reply, marked as answer]
Or provide two separate options, `-features java=1 -features java_pre_jpms=0`

To avoid needing to do this, try removing all files and starting over from a clean checkout.  The configure script should download and configure ACE+TAO and put these settings in its default.features file.

**Reply by lgy-Code** [reply]
> 或者提供两个单独的选项，`-features java=1 -features java_pre_jpms=0`
> 
> 为避免需要执行此操作，请尝试删除所有文件并从干净签出重新开始。configure 脚本应下载并配置 ACE+TAO，并将这些设置放在其 default.features 文件中。

Your guidance was incredibly helpful, thanks a lot!

**Comment by GetIt-now** [comment]
Thanks a lot, I have solved the problem in your way.
